### PR TITLE
Add note on no attestation_data_root

### PR DESCRIPTION
**Pull request by dapplion**
- Address a really old concern raised by https://github.com/ethereum/beacon-APIs/issues/109 where some implementations returned 404 if no aggregate can be built.

### CODE REVIEW

File: apis/validator/aggregate_attestation.yaml [hunk 1]
```diff
@@ -7,6 +7,9 @@ get:
     A 503 error must be returned if the block identified by the response
     `beacon_block_root` is optimistic (i.e. the aggregated attestation attests
     to a block that has not been fully verified by an execution engine).
+
+    A 500 error must be returned if no attestation is available for the requested
+    `attestation_data_root`.
```
Comment (rolfyone) [inline apis/validator/aggregate_attestation.yaml:12]: a 500 would be super weird, why are we going with that?

Comment (dapplion) [reply apis/validator/aggregate_attestation.yaml:12]: The sane alternative is a 404, which is not ideal either.

Comment (rolfyone) [reply apis/validator/aggregate_attestation.yaml]: It's more difinitive, and consistent with what we do when we have routes not found on the states endpoint... I'm happy with that.

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**